server/src/utilities/processFile/processFile.test.ts

```diff
@@ -0,0 +1,59 @@
+import { join } from 'path';
+import Project from 'ts-morph';
+
+import MezzuriteComponent from '../../models/MezzuriteComponent';
+import * as generateComponent from '../generateComponent';
+import * as getComponentType from '../getComponentType';
+import processFile from './processFile';
+
+describe('processFile.ts', () => {
+  const project = new Project({
+    addFilesFromTsConfig: false
+  });
+
+  afterEach(() => {
+    jest.clearAllMocks();
+  });
+
+  it('should resolve the promise with the component data', () => {
+    const filePath = join('.', 'server', 'src', 'utilities', 'processFile', '__mocks__', 'exampleComponent.js');
+    const result: MezzuriteComponent = {
+      checks: {
+        isInstrumented: true
+      },
+      filePath: 'path',
+      name: 'fileName',
+      type: 'react'
+    };
+    Object.defineProperty(getComponentType, 'default', { value: jest.fn(() => 'react') });
+    Object.defineProperty(generateComponent, 'default', { value: jest.fn(() => result) });
+    return processFile(filePath, project).then((component: MezzuriteComponent) => {
+      expect(getComponentType.default).toHaveBeenCalledTimes(1);
+      expect(generateComponent.default).toHaveBeenCalledTimes(1);
+      expect(component).toMatchObject(result);
+    });
+  });
+
+  it('should resolve the promise with a null value when the component data is null', () => {
+    const filePath = join('.', 'server', 'src', 'utilities', 'processFile', '__mocks__', 'exampleNotComponent.js');
+    Object.defineProperty(getComponentType, 'default', { value: jest.fn(() => null) });
+    return processFile(filePath, project).then((component: MezzuriteComponent) => {
+      expect(getComponentType.default).toHaveBeenCalledTimes(1);
+      expect(component).toBeNull();
+    });
+  });
+
+  it('should resolve the promise with a null value when the file path does not exist', () => {
+    const filePath = 'does not exist';
+    return processFile(filePath, project).then((component: MezzuriteComponent) => {
+      expect(component).toBeNull();
+    });
+  });
+
+  it('should resolve the promise with a null value when the filePath is null', () => {
```

Comment:
I'm not sure that I agree with a Promise resolving with `null` when provided an invalid parameter. My first instinct when I think about a Promise-returning function handling invalid parameters is to reject it, as it's not a valid way of calling the function. Thoughts?

Reply:
Agreed. Semantically the promise should reject and leave all responsibility up to the caller. Resolved in https://github.com/Microsoft/vscode-mezzurite/pull/31/commits/792b7100303dcf4fea87143696289b1e87c7d5e4.